### DISCUSSION TITLE
Color detection mechanism does not work if stdout is piped

### DISCUSSION THREAD

**Discussion by pomdtr** [charmbracelet/bubbletea#662] [Q&A] [answered]
I'm trying to build an fzf like cli: the tui output a string to stdout, which is piped to another command.

In order to achieve it, use the `tea.WithOutput(os.Stderr)` option, and the following command `sunbeam | xargs vim`.

However, when using this command, the color is not detected properly:

<img width="654" alt="image" src="https://user-images.githubusercontent.com/17577332/217509518-d7eed564-a570-4bd1-90b5-1ddecbb7778d.png">

<img width="654" alt="image" src="https://user-images.githubusercontent.com/17577332/217509563-ad2d9439-6a13-4fa4-a435-37023f2c96df.png">

**Comment by hopefulTex** [comment, marked as answer]
Hi, you can solve this by forcing a color profile with `lipgloss.SetColorProfile(int)`.
There are 4 modes available, 0:TrueColor, 1:ANSI256, 2:ANSI, 3:Ascii.
This isn't a bug, moreover a precaution the `termenv` lib takes to avoid writing escape sequences when 'not' outputting to a TTY.
